### ISSUE TITLE
Fix reports of errors with the --fix option in declaration-block-trailing-semicolon

### ISSUE THREAD

**Issue by ota-meshi** [stylelint/stylelint#3492]
> Clearly describe the bug

`declaration-block-trailing-semicolon` also reports errors with the `--fix` option.

> Which rule, if any, is the bug related to?

`declaration-block-trailing-semicolon`

> What CSS is needed to reproduce the bug?

e.g.

```css
a { color: pink }
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "declaration-block-trailing-semicolon": "always"
  }
}
```

> Which version of stylelint are you using?

master branch

> How are you running stylelint: CLI, PostCSS plugin, Node API?

execute test case.
I tried `expect(output.results[0].warnings).toEqual([]);` on line 113 of `jest-setup.js`.

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

uses standard.

> What did you expect to happen?

No warnings to be flagged.
The following link says "If context.fix is true, then change root using PostCSS API and return early before report () is called." However, it does not satisfy this specification.
https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#adding-autofixing

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
 FAIL  lib/rules/declaration-block-trailing-semicolon/__tests__/index.js
  ● declaration-block-trailing-semicolon › reject › ["always"] › "a { color: pink }" › single declaration block without trailing semicolon

    expect(received).toEqual(expected)
    
    Expected value to equal:
      []
    Received:
      [{"column": 15, "line": 1, "rule": "declaration-block-trailing-semicolon", "severity": "error", "text": "Expected a trailing semicolon (declaration-block-trailing-semicolon)"}]
    
    Difference:
    
    - Expected
    + Received
    
    - Array []
    + Array [
    +   Object {
    +     "column": 15,
    +     "line": 1,
    +     "rule": "declaration-block-trailing-semicolon",
    +     "severity": "error",
    +     "text": "Expected a trailing semicolon (declaration-block-trailing-semicolon)",
    +   },
    + ]

      111 |                       expect(fixedCode).toBe(testCase.fixed);
      112 |                       expect(fixedCode).not.toBe(testCase.code);
    > 113 |                       expect(output.results[0].warnings).toEqual([]);
```

**Comment by jeddy3** [comment]
@ota-meshi Thanks for the report. It's really well written and clear!
